tribe-ext-calendar-widget-areas/index.php

```diff
@@ -0,0 +1,280 @@
+	 *
+	 * Follow this format to enable a new widget area for each action hook you want
+	 * to use. You will also need to add an additional method toward the end of this
+	 * file for each area added so we know what code to execute.
+	 * 
+	 * @access: public
+	 * @return: array
+	 */
+	public function get_all_areas() {
+		// format is:
+		// action_hook => array of details what to do if we use this hook
+		$areas = array(
+			'tribe_events_before_template'	=>	array(
+				'method'  	=> 'before_template',
+				'name'		=> __( 'TEC Above Calendar', 'tribe-extension' ),
+				'desc'		=> __( 'Widgets in this area will be shown ABOVE The Events Calendar.', 'tribe-extension' ),
+			),				
+			'tribe_events_after_template'	=>	array(
+				'method'  	=> 'after_template',
+				'name'		=> __( 'TEC Below Calendar', 'tribe-extension' ),
+				'desc'		=> __( 'Widgets in this area will be shown BELOW The Events Calendar.', 'tribe-extension' ),
+			),				
+			'tribe_events_before_view'	=>	array(
+				'method'  	=> 'before_view',
+				'name'		=> __( 'TEC Above Single Events', 'tribe-extension' ),
+				'desc'		=> __( 'Widgets in this area will be shown ABOVE Single Events.', 'tribe-extension' ),
+			),				
+			'tribe_events_after_view'	=>	array(
+				'method'  	=> 'after_view',
+				'name'		=> __( 'TEC Above Single Events', 'tribe-extension' ),
+				'desc'		=> __( 'Widgets in this area will be shown BELOW Single Events.', 'tribe-extension' ),
+			),
+		);
+		
+		return apply_filters( 'tribe_ext_calendar_widget_areas', $areas );
+	}
+	
+	/**
+	 * Build options to present to user
+	 * 
+	 * @access: public
+	 * @return: array
+	 */
+	public function get_available_area_options() {
+		$options = array();
+		
+		foreach ( $this->get_all_areas() as $key => $value ) {
+			$options[$key] = $value['name'];
+		}
+		
+		return $options;
+	}
+	
+	/**
+	 * The chosen widget areas to activate/run.
+	 * 
+	 * @access: public
+	 * @return: array
+	 */
+	public function get_enabled_areas_simple() {
+		$all_available = $this->get_all_areas();
+		
+		$enabled_areas = tribe_get_option( $this->option_key_enabled_areas, $all_available );
+		
+		if ( is_array( $enabled_areas ) ) {
+			foreach ( $enabled_areas as $key => $value ) {
+				if ( ! array_key_exists( $value, $all_available ) ) {
+					unset( $enabled_areas[$key] );
+				}
+			}
+		}
+		
+		return $enabled_areas;
+	}
+		
+	/**
+	 * The chosen widget areas to activate/run.
+	 * 
+	 * @access: public
+	 * @return: array
+	 */
+	public function get_enabled_areas_full_details() {
+		$all_available = $this->get_all_areas();
+		
+		$enabled_areas = $this->get_enabled_areas_simple();
+		
+		$return = array();
+		
+		if ( is_array( $enabled_areas ) ) {
+			foreach ( $enabled_areas as $key => $value ) {
+				if ( array_key_exists( $value, $all_available ) ) {
+					$return[$value] = $all_available[$value];
+				}
+			}
+		}
+		
+		// if none selected, return all available
+		if ( empty( $enabled_areas ) || ! is_array( $enabled_areas ) ) {
```

Comment:
I think we can remove the `is_array()` here as well. Improve performance slightly and simplify the code. We could probably just check if `$return` is empty, that might be slightly simpler/more intuitive for readers.